### DISCUSSION TITLE
Cannot find module error

### DISCUSSION THREAD

**Discussion by MahmoudHassanMustafa** [vercel/community#3137] [Help]
### Summary

I've recently used vercel hosting service but I keep getting this error message:

> This Serverless Function has crashed" 

and when I inspected the logs I found an error logs as follows: 

`2023-07-05T20:20:41.422Z	undefined	ERROR	Cannot find module 'src/utils/constants'
Require stack:
- /var/task/src/auth/guards/jwt-auth.guard.js
- /var/task/src/auth/auth.module.js
- /var/task/src/app.module.js
- /var/task/src/main.js
2023-07-05T20:20:41.431Z	undefined	ERROR	Did you forget to add it to "dependencies" in `package.json`?
RequestId: cbcc2e5f-8a9e-419e-9f08-4efad59799e9 Error: Runtime exited with error: exit status 1
Runtime.ExitError`

but "constants" is not a module it's just a file holding my constants names, I checked the letter-casing and everything seems fine.
I can't figure out the reason behind that error, so if you can please help me guys I'll appreciate it much.
Thank you in advance.

### Steps to Reproduce

![image](https://github.com/vercel/community/assets/73262800/444d758c-973c-4490-b2a3-08c3ea5515db)

**### These are the logs...**

![image](https://github.com/vercel/community/assets/73262800/6157e0e1-4957-4a89-a088-98c195be8bf3)
![image](https://github.com/vercel/community/assets/73262800/8d0e50a8-424f-4585-b8f6-8c7c5cfd70af)

**Comment by dmackca** [comment]
I'm getting the same types of errors. Sometimes it's an external module:
```js
import { sql } from '@vercel/postgres';
```
and sometimes it's a local module
```js
import checkAccessKey from './_auth';
```

I say "sometimes" because it seems to be inconsistent. It'll work on one branch but not another, one function but not another, etc. 
I have found a half dozen other issues reporting the same thing, so I'll follow them all and hope for a solution. Adding `"type": "module"` to package.json doesn't seem to help.

**Comment by asfaqehussain** [comment]
Any solution ???

**Comment by dmackca** [comment]
I think you need to use the full file path including the extension. (I was just using .js, so I don't know if typescript transiplation forgives this). 
Adding the .js file extension seemed to make it work reliably. Otherwise I think it works locally with `vercel dev` but not when you run it on the server.

**Comment by denizsult** [comment]
How did you solved this problem?

**Reply by asfaqehussain** [reply]
Yes, are you facing the same ?

**Reply by denizsult** [reply]
Yes, I just deleted the baseUrl: '/' from the tsconfig file, checked the imports that were causing problems, and adjusted the paths.

**Reply by kadyrov11** [reply]
what did u put in baseUrl

**Reply by vtlinh02** [reply]
agree. The problem is that in your local development, you set baseURL = "./" which mean your project like "Your_project/". But in vercel, you set baseURL = "./" which mean "vercel_server/". So it will not find your path.
So, to conclude, there are 2 ways to fix this:
1. Remove the baseURL in your tsconfig.json file
2. Update the baseURL in Vercel.

**Comment by FlacorLopes** [comment]
Relative imports with .js extension seems to be the way to go

import { gateway } from '../../../../src/api-lib/gateway.js' 

instead of
import { gateway } from 'src/api-lib/gateway.ts' or without extension
